### PR TITLE
Use root kmem to detect if kmem limit have been initialized

### PR DESCRIPTION
**Pull request by hqhq**
Alternative of #962
Fixes: #841

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

### CODE REVIEW

**Comment by hqhq** [comment]
ping @vishh @mlaventure @cyphar @mrunalp @dubstack

**Comment by cyphar** [comment]
Does this work properly for cases where `memory.use_hierarchy = 0`? Also, does this properly handle cases where the limit is updated in a parent after we've created the child?

File: libcontainer/cgroups/fs/memory.go [hunk 1]
```diff
@@ -67,12 +65,15 @@ func (s *MemoryGroup) SetKernelMemory(path string, cgroup *configs.Cgroup) error
 		if err != nil {
 			return err
 		}
-		switch system.GetLongBit() {
-		case 32:
-			kmemInitialized = uint32(kmemValue) != uint32(math.MaxUint32)
-		case 64:
-			kmemInitialized = kmemValue != uint64(math.MaxUint64)
+		memcg, err := cgroups.FindCgroupMountpoint("memory")
```
Comment (vishh) [inline libcontainer/cgroups/fs/memory.go]: Will this work when runc is used inside of a container?

**Comment by hqhq** [comment]
@cyphar Yes I think it will work properly for cases where `memory.use_hierarchy = 0`, but for cases that kernel memory will be updated before, if kernel memory limit is set to a number, than updated to `-1` which will reset kernel memory to unlimited, logic in this PR will consider kernel memory as uninitialized. This issue is also existed on the current version.

@vishh Yeah there will be issue when runc is used inside of a container.

This PR is not workable, I'll close, thank you.

**Pull request closed without merging**